skbio/metadata/_interval.py

```diff
@@ -0,0 +1,274 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
```

Comment:
There shouldn't be any name clashing, as different protocols are invoked `thing.member` vs `thing['member']`. But I don't know if it is worth overcomplicating this as the `metadata` is available, and it *is* a mapping.